modules/initrd-fbterm.nix

```diff
@@ -0,0 +1,52 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+with import ./initrd-order.nix;
+
+let
+  device_name = config.mobile.device.name;
+  cfg = config.mobile.boot.stage-1.fbterm;
+  fontsConf = pkgs.writeText "fonts.conf" ''
+    <?xml version="1.0"?>
+    <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+    <fontconfig>
+      <cachedir>/var/cache/fontconfig</cachedir>
+      <dir>${pkgs.terminus_font}</dir>
+    </fontconfig>
+  '';
+in
+{
+  options.mobile.boot.stage-1.fbterm = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Enables fbterm.
+      '';
+    };
+    fb = mkOption {
+      type = types.str;
+      default = "/dev/fb1";
+      description = ''
+        framebuffer to run fbterm on.
+      '';
+    };
+  };
+
+  config.mobile.boot.stage-1 = lib.mkIf cfg.enable {
+    init = lib.mkOrder AFTER_FRAMEBUFFER_INIT ''
+      (
+      touch /init.log
+      echo "Starting fbterm!"
+      env FB=${cfg.fb} fbterm -n terminus -s 32 </dev/tty1 -- tail -n 200 -f /init.log &
+      echo "Started fbterm!"
```

Comment:
Hard-mode: could we get a real console (vt?) attached so that an attached keyboard can work with it? Thinking about breaking to a shell in initrd, allowing keyboard input on-device, in case there's issues with adb and rndis.

Reply:
Yes, that should be possible (drivers permitting) and I'll give it a shot at some point. Though maybe that's material for a follow-up PR?

Reply:
Definitely for a future PR!